### PR TITLE
Anna/test project

### PR DESCRIPTION
**Pull request by Atomyka**
Further unit tests for default_unit.

### CODE REVIEW

**Comment by Roger-luo** [comment]
CI fails

**Review by Roger-luo — CHANGES_REQUESTED** [review]

File: lib/BloqadeExpr/test/units.jl [hunk 1]
```diff
@@ -10,7 +10,12 @@ using Unitful: kHz, µm
 
     h = RydInteract(atoms = [(1,), (2,), (3,), (4)], C = 109.2kHz * µm^6)
     @test h.C ≈ 0.1092
-
     h = rydberg_h([(1,), (2,)], C = 109.2kHz * µm^6)
     @test h.C ≈ 0.1092
 end
+
+@testset "further units" begin
```
[the remark this answers is not in the document]
Comment (Atomyka) [reply lib/BloqadeExpr/test/units.jl]: Ok! I put them in the above testset since they are in the style of the previous "Hamiltonian terms"

**Comment by Atomyka** [comment]
> CI fails

Ah, I think I was missing
 using BloqadeExpr: default_unit.

File: lib/BloqadeExpr/test/units.jl [hunk 2]
```diff
@@ -10,7 +10,12 @@ using Unitful: kHz, µm
...
+    @test SumOfXPhase(5, 2kHz, 2.0).ϕ ≈ 2.0 # Should test line 4, i.e. NoUnits version of default_unit
```
[the remark this answers is not in the document]
Comment (Atomyka) [reply lib/BloqadeExpr/test/units.jl]: Good to know!

File: lib/BloqadeExpr/test/units.jl [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
+    @test SumOfX(3, [2μHz,2Hz,2MHz]).Ω ≈ [2*1e-12,2*1e-6,2]
 
     h = RydInteract(atoms = [(1,), (2,), (3,), (4)], C = 109.2kHz * µm^6)
     @test h.C ≈ 0.1092
-
     h = rydberg_h([(1,), (2,)], C = 109.2kHz * µm^6)
     @test h.C ≈ 0.1092
 end
+
+unitless_values = [5, 0:0.1:1, [1,2,3]]
```
[the remark this answers is not in the document]
Comment (Atomyka) [reply lib/BloqadeExpr/test/units.jl:21]: I saw that her PR was merged into the master branch, so I thought I should build on top of that?
Or is that something you would take care of as the administrator?

[the remark this answers is not in the document]
Comment (Atomyka) [reply lib/BloqadeExpr/test/units.jl:21]: I saw that her PR was merged into the master branch, so I thought I should build on top of that?
Or is that something you would take care of as the administrator?

**Comment by Roger-luo** [comment]
close in favor of #381

**Pull request closed without merging**